functions/lib/logging.py

```diff
@@ -0,0 +1,32 @@
+import logging
+
+
+class Logger():
+  def __init__(self, logger_name):
+    self.logger = logging.getLogger(logger_name)
+    self.tags = {}
+
+  def add_tag(self, key, value):
+    self.tags[key] = value
+
+  def add_tags(self, tags):
+    self.tags = self.tags | tags
+
+  def remove_tag(self, key):
+    self.tags.pop(key)
+
+  def remove_all_tag(self):
+    self.tags.clear()
+  
+  def debug(self, message, tags={}) : 
+    self.logger.debug(message, extra={"labels": tags | self.tags})
+
+  def info(self, message, tags={}):
+    self.logger.info(message, extra={"labels": tags | self.tags})
+  
+  def warning(self, message, tags={}):
+    self.logger.warn(message, extra={"labels": tags | self.tags})
```

Comment:
no longer use warn, as it is deprecated, use warning instead

Reply:
yes oops, good suggestion